include/solvers/mpm_base.tcc

```diff
@@ -92,8 +92,39 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
     abort();
   }
 
-  bool vtk_statevar = false;
+  // VTK particle variables
+  // Initialise container with empty vector
+  vtk_vars_.insert(
+      std::make_pair(mpm::VariableType::Scalar, std::vector<std::string>()));
+  vtk_vars_.insert(
+      std::make_pair(mpm::VariableType::Vector, std::vector<std::string>()));
+  vtk_vars_.insert(
+      std::make_pair(mpm::VariableType::Tensor, std::vector<std::string>()));
+
+  if ((post_process_.find("vtk") != post_process_.end()) &&
+      post_process_.at("vtk").is_array() &&
+      post_process_.at("vtk").size() > 0) {
+    // Iterate over vtk
+    for (unsigned i = 0; i < post_process_.at("vtk").size(); ++i) {
+      std::string attribute =
+          post_process_["vtk"][i].template get<std::string>();
+      if (variables_.find(attribute) != variables_.end())
+        vtk_vars_[variables_.at(attribute)].emplace_back(attribute);
+      else {
+        console_->warn(
```

Comment:
Use a bool and avoid repetition

Reply:
I think having repetition is okay. This is just to note which variables are not available in one go.